exercises/bank-account/canonical-data.json

```diff
@@ -0,0 +1,317 @@
+{
+  "exercise": "bank-account",
+  "comments": [
+    "This exercise is a good candidate to practice concurrency, which",
+    "may not be available or possible in the implementing language track.",
+    "In this case, all tests can be executed on a single thread. If there",
+    "are tests that don't work using a single thread, these should be",
+    "marked with a concurrency scenario.",
+    "----------------------------------------------------------------",
+    "For tests which requires multiple expected values/errors",
+    "Should put the expected values/errors at each operation",
+    "And then the last expected value/error should be the case expected",
+    "For errors that are suppose to raise non fatal errors",
+    "Should have an extra field called `fatal` which holds a boolean"
```

Comment:
I don't currently see a use for this - these two lines could be removed.

Maybe you could help understand why these two lines are here, by answering this question:
What should the tests do differently if an error is fatal, versus if it's not?

Reply:
An error that is not fatal should make the program being able to continued to be runned, I guess

Reply:
Ah, I was thinking single-threadedly.

Here was why I didn't understand at first:
In a single thread, of course any error in an operation other than the very last one **has** to be non-fatal, since the test has to keep running the rest of the operations. Therefore the `fatal` field seemed useless. It always has to be `false`.

Here's why it was a misconception:
But if there are multiple threads, you might need to say some errors are fatal (stops that thread) or not. therefore the field is useful.

Okay, I think you helped me gain an understanding. Thank you.